### PR TITLE
Disable assembly on new arches

### PR DESCRIPTION
**Pull request by jeremyd2019**
Arches can be added to case statements once assembly support is added.

Openssl is a little different, and specifically handles arm because it 'needs' to know 32 vs 64-bit.

### CODE REVIEW

**Comment by Biswa96** [comment]
Does `mingw64 aarch64_asm` work in openssl for aarch64?

**Comment by jeremyd2019** [comment]
No.  I believe the reason is that Windows AArch64 uses the Thumb instruction set, while pretty much everybody else uses the ARM instruction set on AArch64.

**Comment by jeremyd2019** [comment]
I had hoped my patch to add new configs to openssl for arm and aarch64 wouldn't be needed, since I was disabling asm anyway, but it turns out that `mingw64` config was passing an argument when building the rc that made it try to use x86_64.

**Comment by Biswa96** [comment]
I forgot to tell that openssl _recently_ changes something to fix AArch64 build. See my upstream issue https://github.com/openssl/openssl/pull/12620

**Comment by Biswa96** [comment]
I forgot to tell that openssl _recently_ changes something to fix AArch64 build. See my upstream issue https://github.com/openssl/openssl/issues/13414

**Comment by jeremyd2019** [comment]
I would certainly welcome anything better, this was just to get things to build for me.  But I found that upstream issue hard to follow - there was a comment about adding aarch64, a reply to open a new pull request, but I didn't see anything more than that.

**Review by mati865 — APPROVED** [review]
